sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -1012,6 +1012,8 @@ public PDone expand(PCollection<T> input) {
         case UNBOUNDED:
           return input
               .apply(MapElements.into(new TypeDescriptor<PubsubMessage>() {}).via(getFormatFn()))
+              // TODO(BEAM-10869): Also plump through PubsubMessageWithAttributesCoder if needed.
+              .setCoder(PubsubMessagePayloadOnlyCoder.of())
```

Comment:
Won't this break pipeline update?

Also, the implementation only uses the payload so I don't think we need the TODO since we don't use any of the other attributes from the message so we can leave a comment here stating instead that the unbounded sink only uses the payload of the pubsub message.

Reply:
That's a good point. We can have it guard with "beam_fn_api" and "use_runner_v2". 

It seems like given Cham work here: https://github.com/apache/beam/pull/12760, we just need to send `PubsubMessage.data` as bytes for portable dataflow.

Reply:
I misunderstood what PR12760 does. Can you explain more about "we don't use any of the other attributes from the message"? It seems like both `PubsubMessage` in python and java contain 'attribute' map.

Reply:
PubsubUnboundedSink only consumes the data blob and sets the other attributes using properties from the element (e.g. timestamp).